doc-bridgepoint/notes/9554_integrity_int.md

```diff
@@ -0,0 +1,311 @@
+model data.  Thus, if model data is not well-linked, then the Prebuilder
+will omit data.  This work is intended to identify exactly these
+types of missing links.  Thus, it is a requirement of the integrity
+checker to load all instances independently of the linkage.  The
+MC-3020 loader does exactly this.  It loads all instances first, and
+then _batch relates_ next.  When a link is not found, the instance
+remains unlinked.  The integrity checker can then detect and report
+the missing link by selecting all instances and exhaustively interrogating
+the linkage.
+
+6.2 Inter-Project References (IPRs)  
+We will need to consider that checking across project boundaries
+means we need to have all of the data from all included projects.
+A multi-select is required.  In the GUI, this will mean selecting
+all inter-dependent projects to run the Check Referential Integrity.
+From the command line, all inter-dependent projects must be passed
+as parameters on the command line.
+
+Alternatively, as discussed on August 9, 2017, we will automatically
+select all projects in the workspace in all cases.  This is the
+new plan of record.
+
+6.3 Command Line  
+Most of the interface to the integrity checker is supplied through
+`xtumlmc_build` passing `xtuml_integrity` as the first argument.
+This keeps the GUI and command line close and simplifies the plugin.
+
+6.4 Global Native Types  
+BridgePoint supplies global, native xtUML types to projects.  These types
+are not stored as part of the project xtUML model data.  Instead, they
+are loaded from within the tool.  This file is found in the org.xtuml.bp.pkg
+project at `globals/Globals.xtuml`.
+
+`xtumlmc_build xtuml_integrity` will automatically include this data along
+with the selected project(s) model data.  An option is provided (`-g`) to
+exclude this data so that Prebuilder output can be used as integrity input.
+
+6.5 Out-and-Back  
+A select out-and-back statement was added to detect when some associations
+may have been over-written with batch relate.  This covers the case of
+an extra subtype that the instance loader and batch relater would allow
+to occur.
+
+6.6 `defaults.xtuml`  
+This file was include in bp.core and is no longer used.  Deleting it.
+
+6.7 Error Counter  
+The number of errors found are counted and included in the output report.
+A counter local to each class type limits the number of errors found of
+one type of instance to 10.  This will prevent a waterfall of errors in
+the report.
+
+6.8 Simplified Plugin  
+The plugin depends on io.image and `System_c` (`S_SYS`).  These dependencies
+are only in place because needing to have the _project_ variable.  This
+can be obtained in simpler ways.
+
+6.9 bp.core ooaofooa  
+See [[2.5]](#2.5).  
+
+6.10 Enhanced Loader  
+Consider fixing the loader to deal with unrecognized data gracefully.  
+Consider enhancing the escher loader to support OOA of Graphics instances.  
```

Comment:
Made this part of the follow-on issue...
https://support.onefact.net/issues/9742